pkg/manifests/manifests_test.go

```diff
@@ -1557,6 +1557,74 @@ func TestPrometheusQueryLogFileConfig(t *testing.T) {
 	}
 }
 
+func TestPrometheusRetentionConfigs(t *testing.T) {
+	for _, tc := range []struct {
+		name                  string
+		retention             string
+		retentionSize         string
+		expectedRetention     string
+		expectedRetentionSize string
+	}{
+		{
+			name:                  "both retention and retentionSize defined",
+			retention:             "30d",
+			retentionSize:         "15GiB",
+			expectedRetention:     "30d",
+			expectedRetentionSize: "15GiB",
+		},
+		{
+			name:                  "only retention defined",
+			retention:             "45d",
+			expectedRetention:     "45d",
+			expectedRetentionSize: "",
+		},
+		{
+			name:                  "only retentionSize defined",
+			retentionSize:         "25GB",
+			expectedRetention:     "",
+			expectedRetentionSize: "25GB",
+		},
+		{
+			name:                  "both retention and retentionSize empty",
+			expectedRetention:     "15d",
+			expectedRetentionSize: "",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			c := NewDefaultConfig()
+			c.ClusterMonitoringConfiguration.PrometheusK8sConfig.Retention = tc.retention
+			c.ClusterMonitoringConfiguration.PrometheusK8sConfig.RetentionSize = tc.retentionSize
+
+			f := NewFactory("openshift-monitoring", "openshift-user-workload-monitoring", c, defaultInfrastructureReader(), &fakeProxyReader{}, NewAssets(assetsPath), &APIServerConfig{}, &configv1.Console{})
+
+			p, err := f.PrometheusK8s(
+				&v1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
+				&v1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
+			)
+
```

Comment:
nit, wdyt about moving the `err != nil`  from lines `1620-1623` to here? Makes the test easier to read no?

Reply:
ya definitely :)

I have updated the code